File: Reservations/Reservations.Domain/Ticket.cs
using Reservations.Domain.Enums;
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Reservations.Domain
{
    public class Ticket : AuditableEntity
    {
        public Ticket()
        {
            TicketExtensions = new HashSet<TicketExtension>();
        }
        public long Id { get; set; }
        public TicketState TicketState { get; set; }
        public decimal Price { get; set; }
        public long? ReservationId { get; set; }
        public Reservation Reservation { get; set; }
        public long? EventOccurenceId { get; set; }
        public EventOccurrence EventOccurence { get; set; }
        public ICollection<TicketExtension> TicketExtensions { get; set; }
    }
}

File: reservation-frontend/src/app/models/ticket.ts
import { EventOccurrence } from "./eventOccurrence"
import { Reservation } from "./reservation"

export class Ticket {
    public id!: number
    public ticketState!: string
    public price!: number
    public reservationId!: number
    public reservation!: Reservation
    public eventOccurrenceId!: number
    public eventOccurrence!: EventOccurrence
}